### DISCUSSION TITLE
Installation process on macOS

### DISCUSSION THREAD

**Discussion by VitaliyYakob** [Netatalk/netatalk#1577] [General]
Hello,

I used this [instruction](https://netatalk.io/stable/htmldocs/compile#build-macos) for deployment, and I have a couple of suggestions for improvement:

1. Do not use wget to get UnicodeData.txt, this is not the default in macOS, and curl is
`curl -O https://www.unicode.org/Public/UCD/latest/ucd/UnicodeData.txt`

2. After completing the "cd build && meson test" step, the user is in the build directory, but in the next step you need to go back to cd..

3. The last action installation script loads and launches the netatakd daemon using outdated approaches
`Running custom install script '/bin/launchctl load -w /Library/LaunchDaemons/io.netatalk.daemon.plist'`

`man launchctl`

>  LEGACY SUBCOMMANDS
>      Legacy subcommands select the target domain based on whether they are executed as root or not. When executed as root, they target the system domain.
> 
>      load | unload [-wF] [-S sessiontype] [-D searchpath] paths ...
>               Recommended alternative subcommands: bootstrap | bootout | enable | disable
> 
>               Load the specified configuration files or directories of configuration files.  Jobs that are not on-demand will be started as soon
>               as possible. All specified jobs will be loaded before any of them are allowed to start. Note that per-user configuration files
>               (LaunchAgents) must be owned by root (if they are located in /Library/LaunchAgents) or the user loading them (if they are located
>               in $HOME/Library/LaunchAgents).  All system-wide daemons (LaunchDaemons) must be owned by root. Configuration files must disallow
>               group and world writes. These restrictions are in place for security reasons, as allowing writability to a launchd configuration
>               file allows one to specify which executable will be launched.

I suggest to consider the possibility of using a modern option

`launchctl bootstrap system/ /Library/LaunchDaemons/io.netatalk.daemon.plist`
`launchctl kickstart system/io.netatalk.daemon`

**Comment by rdmark** [comment]
Thanks for the suggestions. Addressing all of them should be doable. I converted this discussion into two issue tickets;

https://github.com/Netatalk/netatalk/issues/1578
https://github.com/Netatalk/netatalk/issues/1579
